common/src/main/java/org/apache/rocketmq/common/message/MessageExt.java

```diff
@@ -118,18 +118,25 @@ public void setBornHost(SocketAddress bornHost) {
     }
 
     public String getBornHostString() {
-        if (this.bornHost != null) {
-            InetSocketAddress inetSocketAddress = (InetSocketAddress) this.bornHost;
-            return inetSocketAddress.getAddress().getHostAddress();
+        try {
+            if (this.bornHost != null) {
+                InetSocketAddress inetSocketAddress = (InetSocketAddress) this.bornHost;
+                return inetSocketAddress.getAddress().getHostAddress();
+            }
+        } catch (Exception e) {
         }
 
         return null;
     }
 
     public String getBornHostNameString() {
-        if (this.bornHost != null) {
-            InetSocketAddress inetSocketAddress = (InetSocketAddress) this.bornHost;
-            return inetSocketAddress.getAddress().getHostName();
+        try {
+            if (this.bornHost != null) {
+                InetSocketAddress inetSocketAddress = (InetSocketAddress) this.bornHost;
+                return inetSocketAddress.getAddress().getHostName();
+            }
+        } catch (Exception e) {
```

Comment:
It is recommended to check for NPE in advance instead of catching exceptions

Reply:
It would be better to write as this instead of catching exceptions
```java
public String getBornHostNameString() {
        if (this.bornHost != null) {
            InetAddress inetAddress = ((InetSocketAddress) this.bornHost).getAddress();
            return inetAddress != null ? inetAddress.getHostName() : null;
        }
        return null;
    }
```